xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/job/XClassBreakingQuestion.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring.job;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.refactoring.job.question.EntitySelection;
+
+/**
+ * Represents a question asked to the user when he tries to refactor some pages containing used XClass.
+ *
+ * @version $Id$
+ * @since 10.9RC1
+ */
+public class XClassBreakingQuestion
+{
+    /**
+     * The map of entities concerned by the refactoring, where the entity reference if the key.
+     */
+    private Map<EntityReference, EntitySelection> concernedEntities;
+
+    /**
+     * The map of objects that might be impacted by a deletion.
+     * The key is represented by the local name of the page containing the XClass.
+     */
+    private Map<String, Set<String>> impactedObjects;
```

Comment:
I'm really not a fan of this, keep in mind that DeleteJob allow deleting documents from different wikis in the same job. If you want to avoid problem always manipulated references.